src/main/java/teammates/storage/search/SearchManager.java

```diff
@@ -85,6 +86,52 @@ public static void putDocument(String indexName, Document document) {
     }
 
     /**
+     * Batch creates or updates the search documents for the given documents and index.
+     */
+    public static void putDocuments(String indexName, List<Document> documents) {
+        Index index = getIndex(indexName);
+
+        int delay = 2;
+        for (int attempts = 0; attempts < MAX_RETRIES; attempts++) {
+            try {
+                PutResponse result = index.put(documents);
+
+                if (Config.PERSISTENCE_CHECK_DURATION == 0) {
+                    continue;
+                }
+
+                int elapsedTime = 0;
+                boolean isSuccessful = result.getResults().get(0).getCode() == StatusCode.OK;
```

Comment:
This only checks if the first document was put, not the others. You may have to loop through all the results to check that they're all okay? Then perhaps only re-put those that are not okay and check those again?